dbms/src/Interpreters/Context.cpp

```diff
@@ -569,6 +571,19 @@ ConfigurationPtr Context::getUsersConfig()
     return shared->users_config;
 }
 
+void Context::setSecurityConfig(Poco::Util::AbstractConfiguration & config, const LoggerPtr & log)
+{
+    LOG_INFO(log, "Setting secuirty config.");
+    auto lock = getLock();
+    shared->security_config = TiFlashSecurityConfig(config, log);
+}
+
+TiFlashSecurityConfig & Context::getSecurityConfig()
```

Comment:
Will the following case happen?
1. Some thread gets the reference to `TiFlashSecurityConfig` from the return value of this function;
2. The above thread start using this config, WITHOUT the protection of the lock;
3. Some other thread sets the security config thus make the original one dangling.